src/databricks/labs/dqx/llm/README.md

```diff
@@ -0,0 +1,394 @@
+
+## ✅ **Key Features**
+
+- **🔧 Completely Optional**: Not activated by default - requires explicit enablement
+- **🤖 Intelligent Detection**: Uses LLM analysis of table schema and metadata
+- **✨ Multiple Activation Methods**: Various ways to enable when needed
+- **🛡️ Graceful Fallback**: Clear messaging when dependencies unavailable
+- **⚡ Performance Optimized**: Lazy loading and conditional execution
+- **🔍 Duplicate Validation**: Optionally validates detected PKs for duplicates
+- **📊 Confidence Scoring**: Provides confidence levels and reasoning
+- **🔄 Retry Logic**: Handles cases where initial detection finds duplicates
+
+## 📦 **Installation**
+
+### **LLM-Enhanced Usage**
+```bash
+# Install DQX with LLM dependencies using extras
+databricks labs install dqx[llm]
+
+# Now you can enable LLM features when needed
+from databricks.labs.dqx.config import ProfilerConfig, LLMConfig
+config = ProfilerConfig(llm_config=LLMConfig(enable_pk_detection=True))
+```
+
+## 🚀 **Usage Examples**
+
+### **Method 1: Configuration-Based**
+```python
+from databricks.labs.dqx.config import ProfilerConfig, LLMConfig
+from databricks.labs.dqx.profiler.runner import ProfilerRunner
+
+# Default configuration (LLM features disabled)
+config = ProfilerConfig()
+
+# Enable LLM-based PK detection via configuration
+config = ProfilerConfig(
+    llm_config=LLMConfig(
+        enable_pk_detection=True,
+        pk_detection_endpoint="databricks-meta-llama-3-1-8b-instruct"
+    )
+)
+
+runner = ProfilerRunner(ws, spark, installation, profiler, generator)
+checks, summary_stats = runner.run(InputConfig(location="catalog.schema.table"), config)
+
+# Check results
+if "llm_primary_key_detection" in summary_stats:
+    pk_info = summary_stats["llm_primary_key_detection"]
+    print(f"Detected PK: {pk_info['detected_columns']}")
+    print(f"Confidence: {pk_info['confidence']}")
+```
+
+### **Method 2: Options-Based**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Enable via options parameter
+summary_stats, dq_rules = profiler.profile_table(
+    "catalog.schema.table",
+    options={
+        "llm": True,  # Simple LLM enablement
+        "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    }
+)
+
+# Or use the explicit flag
+summary_stats, dq_rules = profiler.profile_table(
+    "catalog.schema.table",
+    options={"enable_llm_pk_detection": True}
+)
+```
+
+### **Method 3: Convenience Method**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Use the convenience method - automatically enables LLM detection
+summary_stats, dq_rules = profiler.profile_table_with_llm_pk_detection(
```

Comment:
please remove Method 3, we already provide method to profile the table with pk detection (method 2).